packages/router/src/util.ts

```diff
@@ -124,6 +129,14 @@ export const matchPath = (
     ...pathname.matchAll(new RegExp(`^${typeMatchingRoute}$`, 'g')),
   ]
 
+  if (matchChildRoutes) {
+    const matchesWithRoutes = [
+      ...pathname.matchAll(new RegExp(`^${typeMatchingRoute}/.+$`, 'g')),
+    ]
+
+    return { match: matchesWithRoutes.length > 0, params: {} }
+  }
+
```

Comment:
I think we should continue the code execution here and also return matched params.

Haven't thought this true, so might not work, but perhaps we can just construct different regexes depending on if `matchChildRoutes` is true or false, and then use that regex in `matchAll` and always save the result to `matches`. Do you think that would work?

Reply:
I think that's basically, what  I've tried first and couldn't get to work.

As it crashes below, due to not being able to find params anymore in that case.

Reply:
I'll have a closer look later today

Reply:
```javascript
  const matchRegex = matchChildRoutes
    ? new RegExp(`^${typeMatchingRoute}/.+$`, 'g')
    : new RegExp(`^${typeMatchingRoute}$`, 'g')

  // Does the `pathname` match the route?
  const matches = [...pathname.matchAll(matchRegex)]
```

I haven't tried it on real code, but it passes all our tests (including you new test). So if something is wrong with that code we need better tests 😉

Reply:
Could have sworn I had the exact same code, but I won't complain :)

Reply:
Changed btw